### PR TITLE
Fix Iris CI script

### PR DESCRIPTION
**Pull request by SkySkimmer**
Also add nice error message when the grep produces an empty result

### CODE REVIEW

**Review by ejgallego — APPROVED** [review]
CI will tell.

**Comment by Zimmi48** [comment]
CI failed:
```
+ tar xfz 1c8c5a553f4df7398bf2c3fd2fb0a943734e1439.tar.gz --strip-components=1
+ rm -f 1c8c5a553f4df7398bf2c3fd2fb0a943734e1439.tar.gz
++ sed 's/.*"dev\.[0-9][0-9.-]*\.\([0-9a-z][0-9a-z]*\)".*/\1/'
++ grep -F '"coq-iris-heap-lang"'
+ iris_CI_REF=dc3b7961
+ '[' -n dc3b7961 ']'
+ exit 1
Makefile.ci:90: recipe for target 'ci-iris' failed
```

File: dev/ci/ci-iris.sh [hunk 1]
```diff
@@ -9,13 +9,15 @@ git_download iris_string_ident
 git_download iris_examples
 
 # Extract required version of Iris (avoiding "+" which does not work on MacOS :( *)
-iris_CI_REF=$(grep -F '"coq-iris"' < "${CI_BUILD_DIR}/iris_examples/coq-iris-examples.opam" | sed 's/.*"dev\.[0-9][0-9.-]*\.\([0-9a-z][0-9a-z]*\)".*/\1/')
+iris_CI_REF=$(grep -F '"coq-iris-heap-lang"' < "${CI_BUILD_DIR}/iris_examples/coq-iris-examples.opam" | sed 's/.*"dev\.[0-9][0-9.-]*\.\([0-9a-z][0-9a-z]*\)".*/\1/')
+[ -n "$iris_CI_REF" ] || { echo "Could not find Iris dependency version" && exit 1; }
```
Comment (RalfJung) [inline dev/ci/ci-iris.sh:13]: Something doesn't seem to work about this combinator... and anyway wouldn't it be more readable to do
```
if [ -z "$iris_CI_REF" ]; then
  echo "Could not find Iris dependency version"
  exit 1
fi
```

Comment (SkySkimmer) [reply dev/ci/ci-iris.sh:13]: It works though https://gitlab.com/coq/coq/-/jobs/845135052

Comment (Zimmi48) [reply dev/ci/ci-iris.sh:13]: I think it's the previous version of the PR which had some issue. This one seems to work fine.

Nonetheless, I agree with the readability concern.

**Comment by SkySkimmer** [comment]
CI passed

**Comment by ejgallego** [comment]
@SkySkimmer wanna address the readability comment ? Otherwise will merge later today.

**Comment by SkySkimmer** [comment]
>@SkySkimmer wanna address the readability comment ?

It looks fine to me.

**Comment by Zimmi48** [comment]
Let's avoid delaying the merge too much given that this is preventing other PRs to have a fully green CI.

**Comment by ejgallego** [comment]
@coqbot: merge now

**Pull request merged**